addons/web/static/src/js/views/sample_server.js

```diff
@@ -116,8 +116,12 @@ odoo.define('web.SampleServer', function (require) {
                 case 'read':
                     return this._mockRead(params);
             }
-            const mock = SampleServer.mockRegistry.get(params.model);
-            const mockFunction = mock && mock[params.method || params.route];
+            // this rpc can't be mocked by the SampleServer itself, so check if there is an handler
```

Comment:
split in several commit: one for registry, one for other fixes/improvements with the tests below